common/persistence/client/factory.go

```diff
@@ -18,7 +18,7 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package persistence
+package client
```

Comment:
Can we rename the `New()` function in this file to maybe `NewFactory()`? I think `client.New()` returning a persistence factory a little bit counter-intuitive.

Reply:
done